src/articles/infrastructure/persistence/relational/mappers/article.mapper.ts

```diff
@@ -23,6 +23,12 @@ export class ArticleMapper {
         (tagEntity: TagEntity) => tagEntity.name,
       );
     }
+    if (raw.claps) {
+      domainEntity.totalClaps = raw.claps.reduce(
+        (sum, clap) => sum + clap.counter,
+        0,
+      );
+    }
```

Comment:
I noticed that in this mapping, totalClaps is already being calculated from raw.claps using a reducer. Given this, could you clarify why we still need to fetch claps for each article separately in the service method? If the claps count is already aggregated here, we might be able to streamline the code and avoid additional database calls. 
Let me know if I’m missing any specific reasoning behind this approach!